test/javascripts/turbolinks_test.coffee

```diff
@@ -389,6 +389,17 @@ describe 'Turbolinks', ->
         assert.include(testDocument.body.textContent, 'Hi bob')
         done()
 
+    it 'returns a promise resolved with a list of nodes that are new (freshly replaced)', (done) ->
```

Comment:
Seems like we could use a few more tests for other codepaths that should resolve as promises with the nodes.